bpo-40838 document `getdoc`, `getmodule`, `getsourcefile` in `inspect` returns `None` (#30575)
python/cpython: Doc/library/inspect.rst

   If the documentation string for an object is not provided and the object is
   a class, a method, a property or a descriptor, retrieve the documentation
   string from the inheritance hierarchy.

>    Might return ``None`` for invalid docstrings.

Reviewer: This method returns `None` if the object doesn't have a docstring. I don't think a docstring even could be invalid.

I'd actually prefer this method returning an empty string if the object doesn't have a docstring. Needing to use `doc = getdoc(x) or ''` everytime is annoying but I can understand changing this would be a backwards incompatible change.

Author: I've updated this to be: ` Might return None for missing or invalid docstrings`

I mean this case by `invalid`:

```python
    if not isinstance(doc, str):
        return None
```

>    Might return ``None`` for missing or invalid docstrings.

Reviewer 2: ```suggestion
   Return ``None`` if the documentation string is invalid or missing.
```

Author: Thank you! 👍

.. function:: getmodule(object)

>    Try to guess which module an object was defined in. Might return ``None``

Reviewer 2: ```suggestion
   Try to guess which module an object was defined in. Return ``None``
```

>    in case of an error.

Reviewer 3: Not necessarily an error; for example it returns None for primitive objects like ints that don't have a `__module__`.
```suggestion
   if the module cannot be determined.
```

.. function:: getsourcefile(object)

   Return the name of the Python source file in which an object was defined

>    or ``None`` if no way can be identified to get the source.  This

Reviewer 4: ```suggestion
   or ``None`` if it cannot be identified.  This
```

Author: Thanks! But, I've copied this exact piece from `inspect.py`, I think it is better to keep it in sync (at least try to do that) 🙂
